project/BRANCHES-AND-TAGS.md

```diff
@@ -0,0 +1,35 @@
+Branches and tags
+=================
+
+Note: details of the release process for the Engine are documented in the
+[RELEASE-CHECKLIST](https://github.com/docker/docker/blob/master/project/RELEASE-CHECKLIST.md).
+
+# Branches
+
+Docker projects should normally have only three living branches at all time, including the regular
+`master` branch:
+
+## `docs` branch
+
+The `docs` branch support documentation updates between product releases. This branch allow us to
+decouple documentation releases from product releases.
+
+## `release` branch
```

Comment:
Generally, documentation fixes. Having a branch for each minor let's you do security releases going back several versions. They don't actually need to exist unless a point release is actually made, especially if the tag is made from master.

The main benefit is that you don't need to force push over the release branch after each release if the the release branch was made with a cherry pick or rebase. Each release becomes a fresh branch off master.